src/shared/edit-util.c

```diff
@@ -104,37 +105,33 @@ int edit_files_add(
         return 1;
 }
 
-static int create_edit_temp_file(
-                const char *target_path,
-                const char *original_path,
-                char * const *comment_paths,
-                const char *marker_start,
-                const char *marker_end,
-                char **ret_temp_filename,
-                unsigned *ret_edit_line) {
-
+static int create_edit_temp_file(EditFile *e) {
         _cleanup_free_ char *temp = NULL;
```

Comment:
One more thing I overlooked in my PR: It's better to use `unlink_and_freep` here in case that the temp file has been created but not assigned to `e->temp` due to errors.

Reply:
Indeed. Nice catch. Addressed.